store/CSDK/src/CarbonReader.cpp

```diff
@@ -79,21 +78,19 @@ jobject CarbonReader::build() {
     jmethodID buildID = jniEnv->GetMethodID(carbonReaderBuilderClass, "build",
         "()Lorg/apache/carbondata/sdk/file/CarbonReader;");
     carbonReaderObject = jniEnv->CallObjectMethod(carbonReaderBuilderObject, buildID);
+    jclass carbonReader = jniEnv->GetObjectClass(carbonReaderObject);
+    hasNextID = jniEnv->GetMethodID(carbonReader, "hasNext", "()Z");
```

Comment:
please move this hasNextID to CarbonReader::hasNext().
if not null then only assign (only one time assignment)

Also same comment for readNextRowID.

Reply:
hasNextId is for method of class, only need one times.
If move to haCarbonReader::hasNext(),  it will increase read carbon file time

Reply:
ok, done